extensions/typescript-language-features/src/languageFeatures/fileConfigurationManager.ts

```diff
@@ -205,6 +208,18 @@ export default class FileConfigurationManager extends Disposable {
 			default: return this.client.apiVersion.gte(API.v333) ? 'auto' : undefined;
 		}
 	}
+
+	private getAutoImportFileExcludePatternsPreference(config: vscode.WorkspaceConfiguration, workspaceFolder: vscode.Uri | undefined): string[] | undefined {
+		return workspaceFolder && config.get<string[]>('autoImportFileExcludePatterns')?.map(p => {
+			// Normalization rules: https://github.com/microsoft/TypeScript/pull/49578
+			const slashNormalized = p.replace(/\\/g, '/');
+			const isRelative = /^\.\.?($|\/)/.test(slashNormalized);
+			return path.isAbsolute(p) ? p :
+				p.startsWith('*') ? '/' + slashNormalized :
+					isRelative ? vscode.Uri.joinPath(workspaceFolder, p).fsPath :
```

Comment:
Should we use `slashNormalized` instead of `p` here?

Reply:
I’m not sure; does it matter to `Uri.joinPath`? I intentionally left it as written since `fsPath` is going to return something platform-specific anyway (it doesn’t matter what we send to TS; I only normalized at all to make checking `isRelative` easier and to avoid constructing a path with both kinds of slashes).